exporter/collector/metrics.go

```diff
@@ -1227,7 +1240,38 @@ func (m *metricMapper) gaugePointToTimeSeries(
 				extraLabels,
 			),
 		},
-	}}
+	})
+
+	// if an untyped prometheus metric from ops agent, double-export as gauge and cumulative
+	// to match GMP exporter behavior. Only for internal use.
+	if untypedDoubleExportFeatureGate.IsEnabled() {
+		val, ok := point.Attributes().Get(GCPOpsAgentUntypedMetricKey)
+		if ok && val.AsString() == "true" {
+			metricKind := metricpb.MetricDescriptor_CUMULATIVE
+			value, valueType := m.numberDataPointToValue(point, metricKind, metric.Unit())
+			series = append(series, &monitoringpb.TimeSeries{
+				Resource:   resource,
+				Unit:       metric.Unit(),
+				MetricKind: metricKind,
+				ValueType:  valueType,
+				Points: []*monitoringpb.Point{{
+					Interval: &monitoringpb.TimeInterval{
+						EndTime: timestamppb.New(point.Timestamp().AsTime()),
```

Comment:
Just wanted to confirm this, but not setting the start time will make this normalize the same way GMP does with these counters? 

That is, when the value goes down, treat it as a reset?

Reply:
I changed this, but it's still doing basically the same thing in the new code so to answer your question: yes it should. This is the normalization function: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/a19d9eab7eac8a63dcf2bf443b481b48ac62570e/exporter/collector/internal/normalization/standard_normalizer.go#L261 (called [from here on a Sum](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/a19d9eab7eac8a63dcf2bf443b481b48ac62570e/exporter/collector/metrics.go#L1158), which the updates I just pushed use for the Cumulative instead of a Gauge)

You can see this in the fixtures, where I had to add [2 new untyped input data points](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/4a74e8973dfd0ec128deba4923c077ee90b30127/exporter/collector/integrationtest/testdata/fixtures/metrics/google_managed_prometheus_expect.json#L96-L162) just to get [one output unknown cumulative](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/4a74e8973dfd0ec128deba4923c077ee90b30127/exporter/collector/integrationtest/testdata/fixtures/metrics/google_managed_prometheus_untyped_expect.json#L530-L562). This is because the first data point gets normalized as a reset point and dropped by our exporter.

As a side note, I had to update our fixture generators to test this because they currently normalize all timestamps to be rightNow() (in order to work with testing against the actual GCM api). But I needed to bypass that for this test to get the cumulative normalization to show up.